src/ukify/ukify.py

```diff
@@ -64,6 +64,15 @@ def shell_join(cmd):
     return ' '.join(shlex.quote(str(x)) for x in cmd)
 
 
+def path_is_readable(s: str | None) -> pathlib.Path | None:
+    """Convert a filename string to a Path and verify access."""
+    if s is None:
+        return None
+    p = pathlib.Path(s)
+    p.open().close()
```

Comment:
hmm, wha? is there no equivalent to access(R_OK)?

Reply:
There is, but it doesn't make sense. This generates a proper exception when there is no access, while `os.access()` requires postprocessing.

Reply:
assert isn't a thing?

Reply:
`assert` is a thing, but unfortunately when python is called with the optimize flag `-O` these are dropped during bytecode compilation and at least Arch used to call setuputils-based python packages with `--optimize=1`, which should have the same effect; not really applicable here, but a general example for this being used in the wild so asserts are not super safe for anything but debug purposes.

Reply:
`assert` is a tool to check internal code correctness or consistency. This check is about external state.